### ISSUE TITLE
Adding support of leak system Neptun

### ISSUE THREAD

**Issue by stalkerj** [0x5e/homebridge-tuya-platform#235]
Hello. We have russian leak system Neptun which latest model works on Tuya platform. We are selling it and we have a large community who use it. We kindly ask you to make a support of it. How can I help you with this?

**Comment by 0x5e** [comment]
is it kind of a water detect device or gas alarm device? do you have more informations of the device? these two category are already supported.

**Comment by stalkerj** [comment]
Its a water detect device.
I got such error in logs:
![Uploading B54648BB-0A10-4DEB-880E-F987231545C3.jpeg…]()

**Comment by stalkerj** [comment]
![6D7A1524-15C7-4AA0-A15C-11B88DDB8337](https://user-images.githubusercontent.com/17808225/218940168-65b861eb-4591-428d-bcc2-7e19046a5be2.jpeg)

**Comment by 0x5e** [comment]
https://github.com/0x5e/homebridge-tuya-platform/blob/develop_1.7.0/SUPPORTED_DEVICES.md
`sfkzq` means water valve, water detect should be `sj`.
if you are just a single user, you can use plugin config to override the product category code and schema.
but if you are selling it, every user need to do this otherwise the plugin can't recognize it.
I suggest you can submit a ticket to tuya and try if they can correct them on the cloud side.

**Comment by stalkerj** [comment]
Our system has a controller with connected water valves and water detectors

**Comment by stalkerj** [comment]
Here is how i got it in Homekit and Tuya app.
In tuya app 1 - valves control, 2 - water counts, whoch can count how many water (in m3) was used. And also there are water leak detectors which shows in alert
![84BEFBF0-A223-4683-A114-B0AB48582E1B](https://user-images.githubusercontent.com/17808225/218944354-e7cd6a3d-e8d0-4f0b-9407-88ebc3ae51cb.jpeg)
![5C25E682-4DD9-4B03-9FB1-E59D2B49E520](https://user-images.githubusercontent.com/17808225/218944377-92dd66d2-933d-448d-bdef-27d6ef6e61d8.png)

**Comment by 0x5e** [comment]
water valve need to have at least one schema with code `switch` or `switch_1`, and for multiple valve `switch_2` `switch_3` ...

I don't know what are your code now.
there's at least two issue:
1. plugin can't get any valve schema, because cloud side just hide them. only standard schema will be shown by default. if you are the owner of the product (I mean on the tuya platform), you'd better submit a ticket to tuya to get help, they may able to modify this, with product owner's permission.
2. but even the first issue get solved, we still can't get leak sensor's schema from cloud api. tuya doesn't support combination products like this. ask them if they have a solution.

when cloud api can return all your product's schemas (valve 1, valve 2, sensor 1, sensor 2), then I can do something to make plugin support your product. good luck.

**Comment by stalkerj** [comment]
Do we have a possibility to correct it locally for my device only? And anyway we will make a ticket in tuya platform because we need it for our buyers.

**Comment by 0x5e** [comment]
Yes, change device control mode on tuya platform, then you can use advanced options to adjust dp code or category code. see https://github.com/0x5e/homebridge-tuya-platform/blob/develop_1.7.0/ADVANCED_OPTIONS.md

But currently plugin only support multiple water valve or single water detector at one time. the plugin aims to cover standard tuya products. for the combination products, I will think about it in the next major version.

**Comment by stalkerj** [comment]
> Yes, change device control mode on tuya platform, then you can use advanced options to adjust dp code or category code. see https://github.com/0x5e/homebridge-tuya-platform/blob/develop_1.7.0/ADVANCED_OPTIONS.md
> 
> But currently plugin only support multiple water valve or single water detector at one time. the plugin aims to cover standard tuya products. for the combination products, I will think about it in the next major version.

If i will change device control mode will it break something in tuya app or it will impact only on to this plugin?

**Comment by 0x5e** [comment]
It only impact cloud api based tuya plugin, homebridge/homeassistant etc.
